extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -196,12 +202,81 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 				}
 			}
 		});
-
 		this._assessmentResults = {
 			issues: serverLevelAssessments,
 			databaseAssessments: databaseLevelAssessments
 		};
 
+		try {
+			const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(connectionProfile.providerId, azdata.DataProviderType.QueryProvider);
+			const query = 'SELECT name, compatibility_level FROM sys.databases; ';
+			const results = await queryProvider.runQueryAndReturn(await (azdata.connection.getUriForConnection(this.sourceConnectionId)), query);
+			const compatLevelHashMap: Map<string, string> = new Map();
+			results.rows.forEach(r => {
+				compatLevelHashMap.set(r[0].displayValue, r[1].displayValue);
+			});
+
+			sendSqlMigrationActionEvent(
+				TelemetryViews.MigrationWizardTargetSelectionPage,
+				TelemetryAction.ServerAssessment,
+				{
+					'sessionId': this._sessionId,
+					'tenantId': this._azureAccount.properties.tenants[0].id,
+					'accountId': this._azureAccount.key.accountId,
+					'hashedServerName': hashString((connectionProfile.serverName))
+				},
+				{
+					'issuesCount': this._assessmentResults.issues.length,
+					'duration': endTime - startTime,
+					'databaseCount': this._assessmentResults.databaseAssessments.length,
+					'startTime': startTime,
+					'endTime': endTime,
+				}
+			);
+
+			this._assessmentResults.issues.forEach(i => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.ServerAssessmentIssues,
+					{
+						'sessionId': this._sessionId,
+						'rule': i.checkId
+					},
+					{}
+				);
+			});
+
+			this._assessmentResults.databaseAssessments.forEach(d => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.DatabaseAssessment,
+					{
+						'sessionId': this._sessionId,
+						'hashedDatabaseName': hashString(d.name),
+						'compatibilityLevel': compatLevelHashMap.get(d.name)!,
+					},
+					{
+						'warningCount': d.issues.length,
+					}
+				);
+
+				d.issues.forEach(w => {
```

Comment:
sending telemetry for each database assessed and each issue seems a bit verbose.  is it possible to answer all the KPI's with an aggregate summary instead?

Reply:
I was under the assumption that we had to track every database getting assessed. This seemed like the best way to do it. Should I just add all the unique issues/warnings with their counts?

Reply:
It'd be strongly preferred to aggregate counts whenever possible